[IMP] mrp: BOM overview (#17768)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/basic_setup/bill_configuration.rst

Reviewer: Add the icon in front of 'Bill of Materals' (it's a beaker or erlenmeyer flask)

Reviewer: ```suggestion
In the *Components* tab of a |BOM|, specify components used to manufacture the product by
```
Tabs get italics now

Reviewer: ```suggestion
:guilabel:`Create "(component) "` option to quickly add the line item, or the :guilabel:`Create and edit`
```
I think that is correct - or you cna eliminate the quotes, just looks odd with empty quotes =)

Reviewer: I don't think this image is necessary - you can keep it or not

Reviewer: If no operations are there, the button is "Add Operatoin" - I would mention that adding the firt operation you click that, and 'add a line' is for additional operations.

Reviewer: ```suggestion
  work center by typing the name and selecting the :guilabel:`Create "(work center) "` option.
```
Or however you want to address the " "

Reviewer: ```suggestion
use the operation's time tracker, which enables the :guilabel:`last (#) work orders` field. This
ensures the time to complete will be estimated based on a specified number of the previous
```

Reviewer: ```suggestion
In the *Instructions* tab, click the :guilabel:`Add a line` link to open the *Create Quality Point* window. Individual steps in
```

Reviewer: I suggest adding a colon and then making the line a sentence.

Ex:
Instructions: This check provides specific....

Reviewer: ```suggestion
Use the *Worksheet* tab to upload a supplementary PDF that will display when the operator
```

Reviewer: ```suggestion
Production`, :guilabel:`Worksheet`, or :guilabel:`Spreadsheet`, the *Message if Failure*
```

Reviewer: I recommend moving this to the last of the text in this section, but above the admonition blocks

Author: This image is specifically for creating a QCP/instruction. I will leave it here for now.

Reviewer: ```suggestion
To locate all |BOMs| that use the affected component, click into the search bar and filter
```
we cna just refer to this as "the search bar" in a generic way

Reviewer: We typically don't show image slike this - I think this can be rmeoved

Reviewer: ```suggestion
Then, select the |BOMs| to update and click :icon:`fa-cog` :menuselection:`(Actions) --> Show BoM
Lines`. The component lines display on the *BoM Lines* page.
```

Reviewer: ```suggestion
To update the component across multiple |BOMs|, first filter by the component again by clicking into
the search bar and selecting :guilabel:`Custom Filter`. Filter by
:guilabel:`Component`, set :guilabel:`is equal to`, and select the component, then click
```

Reviewer: There are a lot of image sin this section - I think the iflters are not necesary to show, I am on the fence about the list, too. This sectoin may not need any images, honestly.

Reviewer: This can be cut

Reviewer: ```suggestion
To begin, navigate to :menuselection:`Manufacturing app --> Products --> Products`. Click in the search bar and search for the component product, then open its product form.
```

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |MO| replace:: :abbr:`MO (Manufacturing Order)`
.. |MOs| replace:: :abbr:`MO (Manufacturing Orders)`
.. |QCP| replace:: :abbr:`QCP (quality control point)`

> A *bill of materials* (or *BoM* for short) documents specific components and their respective

Reviewer: ```suggestion
A *bill of materials*, or *BoM* for short, documents specific components and their respective
```

To create a |BOM|, go to :menuselection:`Manufacturing app --> Products --> Bills of Materials` and
click :guilabel:`New`.

> Optionally specify a :guilabel:`Product Variant` that this |BOM| applies to.

Reviewer: ```suggestion
First, select a :guilabel:`Product' using the drop-down menu. Optionally specify a :guilabel:`Product Variant` that this |BOM| applies to.
```
However you wat to phrase it, but picking the product is step 1

Specify the :guilabel:`Quantity` of units that are produced with a single |BOM|.

> Optionally specify a :guilabel:`Reference` that can be used to differentiate similar |BOMs| for the

Reviewer: I would add an example maybe - like, when you woudl do this (for different colors maybe? I have no idea!) But an example might be nice to really illustrate this. because I get it fundamentally, but I can't actually think of an example =)

  interactive spreadsheet.

.. important::
   When a step is added to a |BOM| operation, Odoo stores it in the **Quality** app as a |QCP|. It
   is possible to manually create a |QCP| with the :guilabel:`Instructions` check type and even
   assign it to an operation other than manufacturing, like receipts.

In the :guilabel:`Team` field, specify the team that is responsible for the step.

> Use the :guilabel:`Instructions` tab to specify the step's instructions.

Reviewer: ```suggestion
Use the *Instructions* tab to specify the step's instructions.
```

.. tip::

>    Type `/` for a list of formatting options and features, including Odoo AI.

Reviewer: Since you have an image for just the AI, I would expand on this specific example.

> The :guilabel:`Miscellaneous` tab contains more |BoM| configurations to customize the following

Reviewer: I don't see this- does something have to be set for thi to appear? If so, I'd mention it 9could just be my runbot)

Author: I'm honestly not sure why it would disappear. There isn't a setting connected to it. My belief is that it always shows up because that's where you configure lead time, how many days it takes to prepare, etc.

>   .. image:: bill_configuration/component-status.png

Reviewer: This is a bit hard to see - yu can actually set the widest part of the image to 933- I recommend zooming in, adjusting the oclumn widths so it can be as large as posisble on-screen, then take the pic and resize to 933 width

     is consumed.

  .. image:: bill_configuration/component-status.png
     :alt: Show the Component Status field on the manufacturing order dashboard.

- :guilabel:`Version`: Displays the current |BoM| version, visible with the Odoo **PLM** app
  installed for managing |BoM| changes.

> - :guilabel:`Additional Notes`: Additioonal notes that are added to an |MO|. These notes are also

Reviewer 2: ```suggestion
- :guilabel:`Additional Notes`: Additional notes that are added to an |MO|. These notes are also
```

> Click :guilabel:`Add a line`, and fill in the :guilabel:`By-product`, :guilabel:`Quantity`, and

Reviewer: ```suggestion
Once the feature is enabled, add by-products to a |BOM| by clicking the *By-products* tab.
Click :guilabel:`Add a line`, and fill in the :guilabel:`By-product`, :guilabel:`Quantity`, and
```

:guilabel:`Unit`. Optionally, specify a :guilabel:`Produced in Operation` for the by-product.

Optionally, the :guilabel:`Cost Share` percentage of a by-product can also be specified at this
stage. The cost share is the percentage of the final production cost for the by-product, divided by
the quantity produced. The total of all by-products' cost share must be less than or equal to `100`.

> .. example::

Reviewer: Excellent example! But I think they can all have GUI labels sinc they are ON the form, yes?

   The by-product, `Mush`, is created in the `Grind grapes` operation when producing `Red Wine`.

   .. image:: bill_configuration/add-by-product.png
      :alt: Show sample by-product in the BoM.

BOM Overview
============

Each |BOM| form includes a *BOM Overview* that serves as a roadmap for all the |BOMs| and operations
involved in manufacturing a :guilabel:`Quantity` of a product, including routes, times, and costs.

To open it, navigate to :menuselection:`Manufacturing app --> Products --> Bills of Materials`, then

> select a |BOM| from the list. In the |BOM| form, click the :icon:`fa-bars` :guilabel:`BOM Overview`

Reviewer: ```suggestion
select a |BOM| from the list. In the |BOM| form, click the :icon:`fa-bars` :guilabel:`BoM Overview`
```

   following the *Manual Assembly* operation. Furthermore, each `Wood Panel` is produced from 6 `Ply
   Layers` and 2 `Wear Layers`.

   .. image:: bill_configuration/bom-overview-table-example.png
      :alt: The BOM Overview report for a table product.

By default, the following columns are listed in the *BOM Overview*:

- :guilabel:`Product`: The product being manufactured is listed in the top row. In subsequent rows,
  the components required to manufacture the product are listed. If sub-assemblies are required for
  a component, expand the row using the :icon:`fa-caret-right` :guilabel:`(Fold)` icon. If an
  operation is required to manufacture the component or product, it is listed under the
  :guilabel:`Operations` section.
- :guilabel:`Quantity`: The number of units required to manufacture the product is listed.
- :guilabel:`Total Cost`: This is the total cost of the unit, including price, labor, and other
  factors. Costs displayed in grey are read directly from the component's product record (the
  :guilabel:`Cost` field), while costs in regular text are computed by the report. The *BOM

>   Overview* only calculates a cost when it has a sub-|BOM| to expand (using manufactured components)

Reviewer 2: just dropping a note here after our new "Child MOs" smart button threw us in a loop: it looks like our current documentation in this section uses sub-BoMs, which i think is perfectly fine! what makes it confusing is that the product seems to refer to MOs generated from sub-BoMs "child MOs". just rambling here to log these observations somewhere 😂  maybe you can clarify that in the documentation, somewhere, sometime..
